video_sampler/sampler.py

```diff
@@ -99,9 +99,25 @@ def sample(self, video_path: str, subs: str = None) -> Iterable[list[FrameObject
             if self.cfg.keyframes_only:
                 stream.codec_context.skip_frame = "NONKEY"
             prev_time = -10
-            for frame_indx, frame in enumerate(container.decode(stream)):
+            
+            # Confirm end_frame is greater than start_frame
+            if self.cfg.end_frame is not None:
+                if self.cfg.end_frame <= self.cfg.start_frame:
+                    raise ValueError(f"Invalid end_frame: {self.cfg.end_frame}. Must be greater than start_frame")
+            
+            # Seek the starting point of the processing interval
+            container.seek(self.cfg.start_frame, stream=stream)
+            frame_rate = stream.average_rate
+            
+            for frame in container.decode(stream):
                 if frame is None:
                     continue
+                
+                frame_indx = int(frame.pts * frame_rate)
```

Comment:
I think `pts` might be expressed in `time_base` whereas `frame_rate` is expressed in frames/second. So, we need to make sure that the units match.
[`pts` pyav docs](https://pyav.org/docs/develop/api/frame.html?highlight=pts#av.frame.Frame.pts)
[`average_rate` pyav docs](https://pyav.org/docs/develop/api/stream.html?highlight=average_rate#av.stream.Stream.average_rate)